src/lib/Lib/CMakeExecution.cpp

```diff
@@ -0,0 +1,90 @@
+//
+// This is a derivative work. originally part of the LLVM Project.
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2023 Vinnie Falco (vinnie.falco@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdocs
+//
+
+#include "lib/Lib/CMakeExecution.hpp"
+
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+#include <llvm/Support/Program.h>
+
+namespace clang {
+namespace mrdocs {
+
+Expected<std::string>
+getCmakePath() {
+    std::vector<std::string> paths = {
+        "/usr/bin/cmake",
```

Comment:
Except for "Program Files" and "opt" (to which we should fall back), all other directories should already be in PATH. 

If they're not, it's because the user made an explicit effort for them to not be in PATH so we shouldn't mess with that.

Also, it's (unfortunately) somewhat common to find tools in `/usr/opt` instead of only `/usr/local/opt`.